File: Lab2_4/C#/MathApp/MathHelper.cs
using System;

namespace ExpressionLibrary
{
    public class MathHelper
    {
        public static double CalculateLn(double value)
        {
            if (value <= 0)
            {
                throw new ArgumentOutOfRangeException(nameof(value), "Logarithm argument must be positive.");
            }
            return Math.Log(value);
        }
    }
}

File: Lab2_4/C++/MathHelper.cpp
#include "MathHelper.h"
#include <cmath>
#include <stdexcept>

double MathHelper::calculateLn(double value) {
    if (value <= 0) {
        throw std::invalid_argument("Logarithm argument must be positive.");
    }
    return std::log(value);
}